src/main/java/lotto/enums/Rank.java

```diff
@@ -35,20 +36,29 @@ public static Rank getRank(int numberOfMatch) {
                 .orElse(LOSE);
     }
 
-    public Prize getPrize() {
-        return prize;
+    public static Rank getRank(int numberOfMatch, boolean hasBonusNumber) {
+        int rankKey = Objects.hash(numberOfMatch, hasBonusNumber);
+        return Optional.ofNullable(ranks.get(rankKey))
+                .orElse(LOSE);
     }
 
-    public int getMatchingCount() {
-        return matchingCount;
+    public Prize getPrize() {
+        return prize;
     }
 
     public boolean isOverMinWinningRank() {
         return matchingCount >= MIN_WINNING_RANK;
     }
 
+    private int rankKey() {
+        return Objects.hash(matchingCount, hasBonus);
+    }
+
     @Override
     public String toString() {
+        if (hasBonus) {
```

Comment:
```suggestion
        if (this == SECOND) {
```
hasBonus는 2등에서만 true이고 toString을 제외하면 사용하지 않고 있기 때문에 제거하는게 어떨까요?